plugins/source/snyk/client/spec.go

```diff
@@ -0,0 +1,40 @@
+package client
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/pavel-snyk/snyk-sdk-go/snyk"
+)
+
+type Spec struct {
+	// APIKey can be set via spec or via "SNYK_API_KEY" environment variable
+	APIKey string `json:"api_key,omitempty"`
+
+	// Organizations is a list of organizations to fetch information from.
+	// By default will fetch from all organizations available for user.
+	Organizations []string `json:"organizations,omitempty"`
+
+	// EndpointURL is optional parameter to override the API URL for snyk.Client.
+	EndpointURL string `json:"endpoint_url,omitempty"`
+}
+
+const (
+	EnvSnykApiKey = "SNYK_API_KEY"
+)
+
+func (s *Spec) getClient(version string) (*snyk.Client, error) {
+	if len(s.APIKey) == 0 {
+		s.APIKey = os.Getenv(EnvSnykApiKey)
```

Comment:
Can we not read this via the env variable but only support it via `spec` as the spec reader already supports environment variable expansion and file expansion. This way the documentation is consistent with all other plugins.

Reply:
Tailscale & GitHub plugins have this, too. Should we remove this from them as well?

Reply:
> I don't think I found the `dependency` API. Can you point me to that? Without it the plugin doesn't have any use-case as far as Im aware.

Currently there's no Snyk SDK that supports it.
We can either fork the one I'm using and implement it ourselves, or postpone.

Reply:
Yes, we should remove this from tailscale and github. then either fork or use directly but that's the only interesting API in snyk